agent/integration/job_runner_integration_test.go

```diff
@@ -230,7 +280,7 @@ func (t *testAgentEndpoint) server(jobID string) *httptest.Server {
 		case "/jobs/" + jobID + "/finish":
 			rw.WriteHeader(http.StatusOK)
 		default:
-			http.Error(rw, fmt.Sprintf("not found; method = %q, path = %q", req.Method, req.URL.Path), http.StatusNotFound)
+			panic(fmt.Sprintf("not found; method = %q, path = %q", req.Method, req.URL.Path))
```

Comment:
What's this change for?

Reply:
I'm curious about this too. Is it because you need a json error message? I think you should craft one if so.

Reply:
this is a relic from when i was bughunting and flailing a bit; i think my logic was that if an endpoint that we haven't accounted for is hit it's a Very Bad Thing, and i don't think these errors are super well surfaced in the tests???? maybe? but also this doesn't really need to be here, i might just yeet it

Reply:
I think it's better to return a 4xx error in such circumstances and raise an error from the Test method. It will be easier to debug how the route was called then.

Reply:
I guess the error handling could happen entirely in the non-test code. Hopefully, writing code that ignores errors is something we have gotten out of the habit of doing.